tests/virt_autotest/libvirt_host_bridge_virtual_network.pm

```diff
@@ -50,26 +50,15 @@ sub run_test {
     upload_logs "$vnet_host_bridge_cfg_name";
     assert_script_run("rm -rf $vnet_host_bridge_cfg_name");
 
-    my $gi_host_bridge = '';
+    my $mac;
+    my $gate = script_output "ip r s | grep 'default via ' | cut -d' ' -f3";
+    #my $gate = script_output "ip -o -4 addr list br0 | awk \"{print \\\$4}\" | cut -d/ -f1";
     foreach my $guest (keys %xen::guests) {
         record_info "$guest", "HOST BRIDGE NETWORK for $guest";
-        #figure out that used with virtio as the network device model during
-        #attach-interface via virsh worked for all sles guest
-        assert_script_run("virsh attach-interface $guest network vnet_host_bridge --model virtio --live");
-        #Get the Guest IP Address from HOST BRIDGE NETWORK
-        if (get_var("XEN") || check_var("HOST_HYPERVISOR", "xen")) {
-            my $mac_host_bridge = script_output("virsh domiflist $guest | grep vnet_host_bridge|grep -oE \"[[:xdigit:]]{2}(:[[:xdigit:]]{2}){5}\"");
-            script_retry "ip neigh | grep $mac_host_bridge | grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"", delay => 60, retry => 6, timeout => 60;
-            $gi_host_bridge = script_output("ip neigh | grep $mac_host_bridge | grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"");
-        }
-        else {
-            script_retry "virsh domifaddr $guest --source arp | grep vnet0 | grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"", delay => 150, retry => 15, timeout => 150;
-            $gi_host_bridge = script_output("virsh domifaddr $guest --source arp | grep vnet0| grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"");
-        }
-        #Confirm HOST BRIDGE NETWORK
-        assert_script_run("ssh root\@$gi_host_bridge 'ping -c2 -W1 openqa.suse.de'", 60);
-        save_screenshot;
-        assert_script_run("virsh detach-interface $guest bridge --current");
+        $mac = '00:16:3e:32:' . (int(rand(89)) + 10) . ':' . (int(rand(89)) + 10);
+        assert_script_run("virsh attach-interface $guest network vnet_host_bridge --model virtio --mac $mac --live");
```

Comment:
@pdostal 
Do you confirm that virtio devices are not supported in Xen guests. (included PV and FV guests). 
Our SLE product and upstream do not support virtio as model on guest xen system very well, as my understood. 
My suggestion, we'd better to use with the "netfront" during attach interface into Xen Guests via virsh. 
For example:
`
if (get_var('VIRT_AUTOTEST') && (get_var('XEN') || check_var('SYSTEM_ROLE', 'xen') || check_var('HOST_HYPERVISOR', 'xen'))) {
     assert_script_run("virsh attach-interface $guest network vnet_host_bridge --model netfront --mac $mac --live");
}
else {
      assert_script_run("virsh attach-interface $guest network vnet_host_bridge --model virtio --mac $mac --live");
}
`